shared/actions/chat2/index.tsx

```diff
@@ -2631,15 +2783,15 @@ const unsentTextChanged = (state, action) => {
   })
 }
 
-const onGiphyResults = (state, action) => {
+const onGiphyResults = (state, action: EngineGen.Chat1ChatUiChatGiphySearchResultsPayload) => {
```

Comment:
do we really have to type these actions also?

Reply:
Yes. One of the differences in TS is that it won't infer function parameter types based on usage. (except in anonymous inline closures).

example: https://www.typescriptlang.org/play/#src=const%20numbers%20%3D%20%5B1%2C%202%2C%203%2C%204%2C%205%5D%0A%0Anumbers.map(a%20%3D%3E%20a%20%2B%201)%0A%0Aconst%20add1Untyped%20%3D%20(a)%20%3D%3E%20a%20%2B%201%0Aconst%20add1Typed%20%3D%20(a%3A%20number)%20%3D%3E%20a%20%2B%201%0A%0A%0Anumbers.map(add1Untyped)%0Anumbers.map(add1Typed)